### ISSUE TITLE
Incorrect page results when Catalog Sorting = Order

### ISSUE THREAD

**Issue by pondermatic** [gocodebox/lifterlms#853]
### Reproduction Steps

1. Set `Catalog Sorting` to `Order (low to high)`
![course-catalog-settings](https://user-images.githubusercontent.com/5377968/57489045-88e81b00-727a-11e9-89b2-0a4395b3fb20.png)

1. Create more courses than `Courses per page` is set to.

1. Set the `Order` post attribute to 0 (the default) on all courses.

1. View the course catalog, page 1.
![catalog-page-1](https://user-images.githubusercontent.com/5377968/57489238-09a71700-727b-11e9-8571-b647487a698d.png)

1. View the course catalog, page 2.
![catalog-page-2](https://user-images.githubusercontent.com/5377968/57489251-0d3a9e00-727b-11e9-8bcf-efda66a5a7d1.png)

### Expected Behavior

All courses are displayed only once.

### Actual Behavior

Some courses will display on multiple pages while other courses do not appear on any pages.

### Error Messages / Logs

none

### System Information

* WordPress
  * version 5.2 on Windows 10
  * version 5.1.1 on CentOS 7
* LifterLMS version 3.31.0
* MariaDB with InnoDB tables
  * version 10.2.9 on Windows 10
  * version 10.0.38 on CentOS 7

### Suspected Cause

When [`LLMS_Query->pre_get_posts()`](https://github.com/gocodebox/lifterlms/blob/3.31.0/includes/class.llms.query.php#L173) sets the `ORDER BY` clause to `menu_order ASC` and multiple courses have their `menu_order` set to the same value, the sort order is nondeterministic according to [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/limit-optimization.html).

Perhaps a 2nd column should be added to the `ORDER BY` clause as a tie-breaker.

I'd be happy to work on this and submit a Pull Request.

**Comment by thomasplevy** [comment]
@pondermatic double check this isn't a theme/3rd party theme conflict (reduce it to only LifterLMS and a twenty theme).

Screenshot from a test case of my own:

![Courses_–_Page_3_–_A_LifterLMS_Sandbox_and_Courses_–_Page_2_–_A_LifterLMS_Sandbox_and_Courses_–_A_LifterLMS_Sandbox](https://user-images.githubusercontent.com/1290739/57492333-4cfe8700-7275-11e9-8e2d-81efae428394.jpg)

All courses have menu order of 0

I see no duplicates or missing courses.

**Comment by thomasplevy** [comment]
See https://iss-853.myliftersite.com/courses

**Comment by pondermatic** [comment]
I deactivated all plugins except LifterLMS and switch to 2019 theme.
![catalog-page-201](https://user-images.githubusercontent.com/5377968/57493227-b1c3dc80-7289-11e9-947f-512599881e57.png)
![catalog-page-202](https://user-images.githubusercontent.com/5377968/57493231-b38da000-7289-11e9-8028-83176df2b160.png)
No change in page results.

Here are the relevant SQL queries.
`WP_Query->get_posts() Main Query`
```SQL
SELECT SQL_CALC_FOUND_ROWS wp_posts.ID
FROM wp_posts
WHERE 1 = 1
  AND (wp_posts.ID NOT IN (
    SELECT object_id
    FROM wp_term_relationships
    WHERE term_taxonomy_id IN (44, 43))
    AND wp_posts.ID NOT IN (
        SELECT object_id
        FROM wp_term_relationships
        WHERE term_taxonomy_id IN (44, 43)))
  AND wp_posts.post_type = 'course'
  AND (wp_posts.post_status = 'publish'
    OR wp_posts.post_status = 'llms-completed'
    OR wp_posts.post_status = 'llms-active'
    OR wp_posts.post_status = 'llms-expired'
    OR wp_posts.post_status = 'llms-on-hold'
    OR wp_posts.post_status = 'llms-pending-cancel'
    OR wp_posts.post_status = 'llms-pending'
    OR wp_posts.post_status = 'llms-cancelled'
    OR wp_posts.post_status = 'llms-refunded'
    OR wp_posts.post_status = 'llms-failed'
    OR wp_posts.post_status = 'llms-txn-failed'
    OR wp_posts.post_status = 'llms-txn-pending'
    OR wp_posts.post_status = 'llms-txn-refunded'
    OR wp_posts.post_status = 'llms-txn-succeeded'
    OR wp_posts.post_status = 'private')
GROUP BY wp_posts.ID
ORDER BY wp_posts.menu_order ASC
LIMIT 0, 8
```

`_prime_post_caches()`
```sql
SELECT wp_posts.*
FROM wp_posts
WHERE ID IN (894,606,624,644,652,659,666,671)
```

With Page 2, the 1st query's `LIMIT` is changed to `8, 8`
`_prime_post_caches()`
```sql
SELECT wp_posts.*
FROM wp_posts
WHERE ID IN (676,671,666,659,652,644,624,606)
```
Notice ids 624, 644, 652, 659, 666, 671, 606 appear in both of the 2nd queries, but in reverse order.

These results are on my development machine, but I've noticed the same behavior on a production machine.

**Comment by pondermatic** [comment]
I striped the 1st query down to it's bare essentials.
```sql
SELECT wp_posts.ID
FROM wp_posts
WHERE wp_posts.post_type = 'course'
GROUP BY wp_posts.ID
ORDER BY wp_posts.menu_order ASC
LIMIT 0, 8
```
ID
--
894
606
624
644
652
659
666
671

```sql
SELECT wp_posts.ID
FROM wp_posts
WHERE wp_posts.post_type = 'course'
GROUP BY wp_posts.ID
ORDER BY wp_posts.menu_order ASC
LIMIT 8, 8
```
ID
--
676
671
666
659
652
644
624
606

**Comment by thomasplevy** [comment]
Perhaps we'd need a larger db to reliably reproduce... 

What's your suggestion for a fix here?

Adjusting to `ORDER BY menu_order ASC, ID ASC`?

PR welcome.

**Comment by thomasplevy** [comment]
I'm not sure what an end user would consider the expected behavior on a tie break now that I'm thinking about it... As a developer I'd fall back to ID but maybe an end user would be expecting it to alpha sort by title as the default for a tie breaker.

I'll take a vote with the team in the AM

When sorting courses/memberships by menu order should the tie breaker for items with the same menu order be the post ID or the post title?

**Comment by pondermatic** [comment]
> Adjusting to `ORDER BY menu_order ASC, ID ASC`?

This would work. I usually prefer to fallback to the name/title.

It would be cool to let the LMS admin choose the 2nd sort.

**Comment by thomasplevy** [comment]
Nope. This is not a democracy. Let's reduce support and sort by `post_title` as the tiebreak.

**Comment by thomasplevy** [comment]
> > Adjusting to `ORDER BY menu_order ASC, ID ASC`?
> 
> This would work. I usually prefer to fallback to the name/title.
> 
> It would be cool to let the LMS admin choose the 2nd sort.

This is an an area where I'd like to avoid adding additional options. Really if we're sorting by this the end user has opted to choose the order by manually assigning the order. If you want pure alpha sorting, use alpha sorting options, if you want to choose the order you should choose the order and you shouldn't need a second option to choose how to not choose the order.

**Comment by thomasplevy** [comment]
Closed by #855
